01-data-structures/02-stacks-and-queues/mystack/mystack_spec.rb

```diff
@@ -6,80 +6,99 @@
-      expect(stack.stack_length).to eq 0
-      expect(stack.top).to eq nil
+    it "pushes a single item and checks the top of the stack stack height" do
       stack.push("Rob")
       expect(stack.top).to eq "Rob"
-      expect(stack.stack_length).to eq 1
     end
 
-    it "pushes an item on top of the stack" do
-      expect(stack.top).to eq nil
-      expect(stack.stack_length).to eq 0
+    it "pushes 5 items and checks the top of the stack and stack height" do
       stack.push("Rob")
       expect(stack.top).to eq "Rob"
       expect(stack.stack_length).to eq 1
+
       stack.push("Ben")
       expect(stack.top).to eq "Ben"
       expect(stack.stack_length).to eq 2
+
       stack.push("Tim")
       expect(stack.top).to eq "Tim"
       expect(stack.stack_length).to eq 3
+
       stack.push("Jinhai")
       expect(stack.top).to eq "Jinhai"
       expect(stack.stack_length).to eq 4
+
       stack.push("Frank")
       expect(stack.top).to eq "Frank"
       expect(stack.stack_length).to eq 5
-
     end
   end
 
   describe "#pop" do
-    it "pops an item off the top of the stack and returns it" do
-      expect(stack.stack_length).to eq 0
+
+    before do
       stack.push("Rob")
       stack.push("Ben")
       stack.push("Tim")
       stack.push("Jinhai")
       stack.push("Frank")
+    end
 
-      expect(stack.top).to eq "Frank"
-      expect(stack.stack_length).to eq 5
+    it "pops an item off the top of the stack and returns it" do
       item = stack.pop
       expect(item).to eq "Frank"
-      expect(stack.top).to eq "Jinhai"
-      expect(stack.stack_length).to eq 4
+
       item = stack.pop
       expect(item).to eq "Jinhai"
+
       item = stack.pop
+      expect(item).to eq "Tim"
+
       item = stack.pop
+      expect(item).to eq "Ben"
+
       item = stack.pop
-      expect(stack.top).to eq nil
-      expect(stack.stack_length).to eq 0
+      expect(item).to eq "Rob"
     end
   end
 
   describe "#empty?" do
-    it "returns true when the stack is empty" do
-      expect(stack.empty?).to eq true
+
+    context "when stack is initialized" do
+
+      it "returns true" do
+        expect(stack.empty?).to eq true
+      end
     end
 
-    it "returns false when the stack is not empty" do
-      stack.push("Rob")
-      expect(stack.empty?).to eq false
-      stack.push("Ben")
-      stack.push("Tim")
-      stack.push("Jinhai")
-      stack.push("Frank")
-      expect(stack.empty?).to eq false
-      item = stack.pop
-      item = stack.pop
-      item = stack.pop
-      item = stack.pop
-      item = stack.pop
-      expect(stack.empty?).to eq true 
+    context "when stack has multiple items loaded" do
+
+      before do
+        stack.push("Rob")
+        stack.push("Ben")
+        stack.push("Tim")
+        stack.push("Jinhai")
+        stack.push("Frank")
+      end
+
+      it "returns false if stack is not empty, returns true if stack is empty" do
+        expect(stack.empty?).to eq false
+
+        stack.pop
```

Comment:
there are 2 cases in this it, split it up into two its and the second one should be the false one